### ISSUE TITLE
NullReferenceException during Performed - Error in StandAlone

### ISSUE THREAD

**Issue by NicoFrois** [Unity-Technologies/InputSystem#969]
I'm using a MIDI Device to control some particles of the Vissual Effect Graph, I'm getting this error when I try to play buttons of the midi device, I only get the error in the  stand alone version, I don't have any trouble in the editor

The code pointed by the error is in the method "CallActionListeners", I'm assuming there's some troubles with the binding(?)
 
`var listenerCount = listeners.length;
            for (var i = 0; i < listenerCount; ++i)
            {
                try
                {
                    listeners[i](context);
                }
                catch (Exception exception)
                {
                    Debug.LogError(
                        $"{exception.GetType().Name} thrown during execution of '{phase}' callback on action '{GetActionOrNull(ref actionStates[actionIndex])}'");
                    Debug.LogException(exception);
                }
            }`
Any help is welcome!

Here's the log:

> NullReferenceException thrown during execution of 'Performed' callback on action 'Action[/MidiDevice/note077,/MidiDevice1/note077]'
UnityEngine.DebugLogHandler:Internal_Log(LogType, LogOption, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
UnityEngine.InputSystem.InputActionState:CallActionListeners(Int32, InputActionMap, InputActionPhase, InlinedArray`1&) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\Actions\InputActionState.cs:1613)
UnityEngine.InputSystem.InputActionState:ChangePhaseOfAction(InputActionPhase, TriggerState&, InputActionPhase) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\Actions\InputActionState.cs:1538)
UnityEngine.InputSystem.InputActionState:ProcessDefaultInteraction(TriggerState&, Int32) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\Actions\InputActionState.cs:1221)
UnityEngine.InputSystem.InputActionState:ProcessControlStateChange(Int32, Int32, Int32, Double, InputEventPtr) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\Actions\InputActionState.cs:898)
UnityEngine.InputSystem.InputActionState:UnityEngine.InputSystem.LowLevel.IInputStateChangeMonitor.NotifyControlStateChanged(InputControl, Double, InputEventPtr, Int64) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\Actions\InputActionState.cs:779)
UnityEngine.InputSystem.InputManager:FireStateChangeNotifications(Int32, Double, InputEvent*) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\InputManager.cs:2769)
UnityEngine.InputSystem.InputManager:UpdateState(InputDevice, InputUpdateType, Void*, UInt32, UInt32, Double, InputEventPtr) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\InputManager.cs:2956)
UnityEngine.InputSystem.InputManager:UpdateState(InputDevice, InputEvent*, InputUpdateType) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\InputManager.cs:2872)
UnityEngine.InputSystem.InputManager:OnUpdate(InputUpdateType, InputEventBuffer&) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\InputManager.cs:2563)
UnityEngine.InputSystem.LowLevel.<>c__DisplayClass7_0:<set_onUpdate>b__0(NativeInputUpdateType, NativeInputEventBuffer*) (at C:\Users\nicolas.frois\Documents\UnityProjects\VJBand_v3\Library\PackageCache\com.unity.inputsystem@1.0.0-preview.3\InputSystem\NativeInputRuntime.cs:64)
UnityEngineInternal.Input.NativeInputSystem:NotifyUpdate(NativeInputUpdateType, IntPtr)
 
(Filename: C:/Users/nicolas.frois/Documents/UnityProjects/VJBand_v3/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Actions/InputActionState.cs Line: 1613)

**Comment by bolbilabs** [comment]
I am getting a very similar error, except I am getting an IndexOutOfRangeException instead of a NullReferenceException on performed. This only occurs in the Windows standalone, both the Editor and Mac Standalone work just fine.

I am using Input System preview.3 - 1.0.0.

I believe this has to do with issues #359 and #358, which were resolved in 0.2.0-preview. Rather than getting the error on multiple actions maps, I am getting the error as the result of having multiple actions in one single action map. My first action of "Movement" seems to produce no errors. My second action of "Confirm" seems to produce the IndexOutOfRangeException when performed.

Here is a log of my Windows standalone build:

```
IndexOutOfRangeException thrown during execution of 'Performed' callback on action 'Player/Confirm[/Mouse/leftButton]'
UnityEngine.DebugLogHandler:Internal_Log(LogType, LogOption, String, Object)
UnityEngine.DebugLogHandler:LogFormat(LogType, Object, String, Object[])
UnityEngine.Logger:Log(LogType, Object)
UnityEngine.Debug:LogError(Object)
UnityEngine.InputSystem.InputActionState:CallActionListeners(Int32, InputActionMap, InputActionPhase, InlinedArray`1&) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Actions/InputActionState.cs:1613)
UnityEngine.InputSystem.InputActionState:ChangePhaseOfAction(InputActionPhase, TriggerState&, InputActionPhase) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Actions/InputActionState.cs:1538)
UnityEngine.InputSystem.InputActionState:ProcessDefaultInteraction(TriggerState&, Int32) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Actions/InputActionState.cs:1221)
UnityEngine.InputSystem.InputActionState:ProcessControlStateChange(Int32, Int32, Int32, Double, InputEventPtr) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Actions/InputActionState.cs:898)
UnityEngine.InputSystem.InputActionState:UnityEngine.InputSystem.LowLevel.IInputStateChangeMonitor.NotifyControlStateChanged(InputControl, Double, InputEventPtr, Int64) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Actions/InputActionState.cs:779)
UnityEngine.InputSystem.InputManager:FireStateChangeNotifications(Int32, Double, InputEvent*) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/InputManager.cs:2769)
UnityEngine.InputSystem.InputManager:UpdateState(InputDevice, InputUpdateType, Void*, UInt32, UInt32, Double, InputEventPtr) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/InputManager.cs:2956)
UnityEngine.InputSystem.InputManager:UpdateState(InputDevice, InputEvent*, InputUpdateType) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/InputManager.cs:2872)
UnityEngine.InputSystem.LowLevel.InputState:Change(InputDevice, InputEventPtr, InputUpdateType) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/State/InputState.cs:75)
UnityEngine.InputSystem.Pointer:OnStateEvent(InputEventPtr) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Devices/Pointer.cs:239)
UnityEngine.InputSystem.Mouse:OnStateEvent(InputEventPtr) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Devices/Mouse.cs:313)
UnityEngine.InputSystem.Mouse:UnityEngine.InputSystem.LowLevel.IInputStateCallbackReceiver.OnStateEvent(InputEventPtr) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Devices/Mouse.cs:323)
UnityEngine.InputSystem.InputManager:OnUpdate(InputUpdateType, InputEventBuffer&) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/InputManager.cs:2550)
UnityEngine.InputSystem.LowLevel.<>c__DisplayClass7_0:<set_onUpdate>b__0(NativeInputUpdateType, NativeInputEventBuffer*) (at /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/NativeInputRuntime.cs:64)
UnityEngineInternal.Input.NativeInputSystem:NotifyUpdate(NativeInputUpdateType, IntPtr) (at C:\buildslave\unity\build\Modules\Input\Private\Input.cs:120)
 
(Filename: /Users/Navi/davidcoolgame/Library/PackageCache/com.unity.inputsystem@1.0.0-preview.3/InputSystem/Actions/InputActionState.cs Line: 1613)
```

**Comment by stefanunity** [comment]
Hello,

we're going to close all the github issues and will concentrate bug information in the Unity public issue tracker https://issuetracker.unity3d.com/product/unity/issues?project=116&status=1&unity_version=&view=hottest.

If you are still experienceing this issue in the latest package version please do file a bug report as described here https://unity3d.com/unity/qa/bug-reporting
For general feedback and discussions you can go to https://forum.unity.com/forums/new-input-system.103/

Thank you!
